spec/models/generic_announcement_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe GenericAnnouncement, type: :model do
+  describe '.default_scope' do
+    before {
+      create_list(:system_announcement, 3)
+      create_list(:instance_announcement, 3)
+    }
+
+    it 'orders by descending valid_from within type' do
+      system, instance = GenericAnnouncement.valid.partition { |a| a.is_a? SystemAnnouncement }
```

Comment:
Useless assignment to variable - `instance`. Use `_` or `_instance` as a variable name to indicate that it won't be used.